API.md

```diff
@@ -0,0 +1,552 @@
+// ...
+
+// stop libp2p
+await libp2p.stop()
+```
+
+### dial
+
+Dials to another peer in the network and establishes the connection.
+
+`dial(peer, options)`
+
+#### Parameters
+
+| Name | Type | Description |
+|------|------|-------------|
+| peer | [PeerInfo](https://github.com/libp2p/js-peer-info), [PeerId](https://github.com/libp2p/js-peer-id), [multiaddr](https://github.com/multiformats/js-multiaddr), `string` | peer to dial |
+| [options] | `Object` | dial options |
+| [options.signal] | `AbortSignal` | abort signal |
+
+#### Returns
+
+| Type | Description |
+|------|-------------|
+| `Promise<Connection>` | Promise resolves with the [Connection](https://github.com/libp2p/js-interfaces/tree/master/src/connection) instance |
+
+#### Example
+
+```js
+// ...
+const conn = await libp2p.dial(remotePeerInfo)
+```
+
+### dialProtocol
+
+Dials to another peer in the network and selects a protocol to communicate with that peer.
+
+`dialProtocol(peer, protocols, options)`
+
+#### Parameters
+
+| Name | Type | Description |
+|------|------|-------------|
+| peer | [PeerInfo](https://github.com/libp2p/js-peer-info), [PeerId](https://github.com/libp2p/js-peer-id), [multiaddr](https://github.com/multiformats/js-multiaddr), `string` | peer to dial |
+| protocols | `String|Array<String>` | Strings that identifies the protocols (e.g '/ipfs/bitswap/1.1.0') |
+| [options] | `Object` | dial options |
+| [options.signal] | `AbortSignal` | abort signal |
+
+#### Returns
+
+| Type | Description |
+|------|-------------|
+| `Promise<Connection>` | Promise resolves with the [Connection](https://github.com/libp2p/js-interfaces/tree/master/src/connection) instance |
+
+#### Example
+
+```js
+// ...
+const conn = await libp2p.dialProtocol(remotePeerInfo, protocols)
+```
+
+### hangUp
+
+Closes an open connection with a peer, graciously.
+
+`hangUp(peer)`
+
+#### Parameters
+
+| Name | Type | Description |
+|------|------|-------------|
+| peer | [PeerInfo](https://github.com/libp2p/js-peer-info), [PeerId](https://github.com/libp2p/js-peer-id), [multiaddr](https://github.com/multiformats/js-multiaddr), `string` | peer to hang up |
+
+#### Returns
+
+| Type | Description |
+|------|-------------|
+| `Promise<void>` | Promise resolves once connection closes |
+
+#### Example
+
+```js
+// ...
+await libp2p.hangUp(remotePeerInfo)
+```
+
+### handle
+
+Registers a given handler for the given protocols.
+
+`libp2p.handle(protocols, handler)`
+
+#### Parameters
+
+| Name | Type | Description |
+|------|------|-------------|
+| protocols | `Array<String>|String` | protocols to register |
+| handler | `function({ connection:*, stream:*, protocol:string })` | handler to call |
+
+
+#### Example
+
+```js
+// ...
+const handler = ({ connection, stream, protocol }) => {
+  // use stream or connection according to the needs
+}
+
+libp2p.handle('/echo/1.0.0', handler)
+```
+
+### unhandle
+
+Unregisters all handlers with the given protocols
+
+`libp2p.unhandle(protocols)`
```

Comment:
What if I only want to unregister a specific handle?

Reply:
we currently do not support it, but maybe you are right and we should support it. @jacobheun any opinions here for we not supporting it?

Reply:
I will add a note on `handle`